File: tests/algorithms/transpiler/FS/dynamic_programming/fizz_buzz.fs
// Generated 2025-08-13 07:12 +0700

exception Return
let mutable _nowSeed:int64 = 0L
let mutable _nowSeeded = false
let _initNow () =
    let s = System.Environment.GetEnvironmentVariable("MOCHI_NOW_SEED")
    if System.String.IsNullOrEmpty(s) |> not then
        match System.Int32.TryParse(s) with
        | true, v ->
            _nowSeed <- int64 v
            _nowSeeded <- true
        | _ -> ()
let _now () =
    if _nowSeeded then
        _nowSeed <- (_nowSeed * 1664525L + 1013904223L) % 2147483647L
        int _nowSeed
    else
        int (System.DateTime.UtcNow.Ticks % 2147483647L)

_initNow()
let rec _str v =
    let s = sprintf "%A" v
    let s = if s.EndsWith(".0") then s.Substring(0, s.Length - 2) else s
    s.Replace("[|", "[")
     .Replace("|]", "]")
     .Replace("; ", " ")
     .Replace(";", "")
     .Replace("\"", "")
let __bench_start = _now()
let __mem_start = System.GC.GetTotalMemory(true)
let rec fizz_buzz (number: int) (iterations: int) =
    let mutable __ret : string = Unchecked.defaultof<string>
    let mutable number = number
    let mutable iterations = iterations
    try
        if number < 1 then
            ignore (failwith ("starting number must be an integer and be more than 0"))
        if iterations < 1 then
            ignore (failwith ("Iterations must be done more than 0 times to play FizzBuzz"))
        let mutable out: string = ""
        let mutable n: int = number
        while n <= iterations do
            if (((n % 3 + 3) % 3)) = 0 then
                out <- out + "Fizz"
            if (((n % 5 + 5) % 5)) = 0 then
                out <- out + "Buzz"
            if ((((n % 3 + 3) % 3)) <> 0) && ((((n % 5 + 5) % 5)) <> 0) then
                out <- out + (_str (n))
            out <- out + " "
            n <- n + 1
        __ret <- out
        raise Return
        __ret
    with
        | Return -> __ret
ignore (printfn "%s" (fizz_buzz (1) (7)))
ignore (printfn "%s" (fizz_buzz (1) (15)))
let __bench_end = _now()
let __mem_end = System.GC.GetTotalMemory(true)
printfn "{\n  \"duration_us\": %d,\n  \"memory_bytes\": %d,\n  \"name\": \"main\"\n}" ((__bench_end - __bench_start) / 1000) (__mem_end - __mem_start)

File: tests/rosetta/transpiler/Java/fizzbuzz.java
public class Main {

    public static void main(String[] args) {
        {
            long _benchStart = _now();
            long _benchMem = _mem();
            for (int i = 1; i < 101; i++) {
                if (Math.floorMod(i, 15) == 0) {
                    System.out.println("FizzBuzz");
                } else                 if (Math.floorMod(i, 3) == 0) {
                    System.out.println("Fizz");
                } else                 if (Math.floorMod(i, 5) == 0) {
                    System.out.println("Buzz");
                } else {
                    System.out.println(_p(i));
                }
            }
            long _benchDuration = _now() - _benchStart;
            long _benchMemory = _mem() - _benchMem;
            System.out.println("{");
            System.out.println("  \"duration_us\": " + _benchDuration + ",");
            System.out.println("  \"memory_bytes\": " + _benchMemory + ",");
            System.out.println("  \"name\": \"main\"");
            System.out.println("}");
            return;
        }
    }

    static boolean _nowSeeded = false;
    static int _nowSeed;
    static int _now() {
        if (!_nowSeeded) {
            String s = System.getenv("MOCHI_NOW_SEED");
            if (s != null && !s.isEmpty()) {
                try { _nowSeed = Integer.parseInt(s); _nowSeeded = true; } catch (Exception e) {}
            }
        }
        if (_nowSeeded) {
            _nowSeed = (int)((_nowSeed * 1664525L + 1013904223) % 2147483647);
            return _nowSeed;
        }
        return (int)(System.nanoTime() / 1000);
    }

    static long _mem() {
        Runtime rt = Runtime.getRuntime();
        rt.gc();
        return rt.totalMemory() - rt.freeMemory();
    }

    static String _p(Object v) {
        return v != null ? String.valueOf(v) : "<nil>";
    }
}
